packages/web-components/src/components/ic-checkbox/ic-checkbox.tsx

```diff
@@ -141,6 +145,9 @@ export class Checkbox {
 
   disconnectedCallback(): void {
     removeFormResetListener(this.host, this.handleFormReset);
+    this.host
+      .querySelector("ic-text-field")
+      ?.removeEventListener("icChange", (e) => e.stopImmediatePropagation());
```

Comment:
I don't think this will tidyup correctly as it is not removing the exact function that is bound in the addEventlistener, it is being passed a different function instance. 
Think it needs to be more like the example here: https://medium.com/@bigcatplichta/javascript-use-bind-to-dynamically-add-and-remove-event-listeners-d6b443877a73